app/game/game.js

```diff
@@ -95,6 +103,41 @@ const GAME = (humanUsername, aiUsername, gameId) => {
         }
     }
 
+    const onTimeout = (gameId,AI_USERNAME,HUMAN_USERNAME) => {
+        console.log("game-----------------------------------------------end");
+        // Check if both players are logged in
+        if (CLIENTS_HANDLER.areBothPlayersLoggedIn(gameId,AI_USERNAME,HUMAN_USERNAME)) {
+            // Send game end message to both players
+            sendDataToPlayer(gameId, HUMAN_USERNAME, GameEndData('win'));
+            sendDataToPlayer(gameId, AI_USERNAME, GameEndData('win'));
+            // Update player stats
+            CLIENTS_HANDLER.updatePlayerStats(HUMAN_USERNAME, 'win');
+            CLIENTS_HANDLER.updatePlayerStats(AI_USERNAME, 'win');
+            removeGame(gameId);
+        } else {
+            console.log("At least one user is not loggedin");
+            removeGame(gameId);
+        }
+    }
+
+    const onShipExplode = (gameId,AI_USERNAME,HUMAN_USERNAME) => {
```

Comment:
We don't need AI_USERNAME, HUMAN_USERNAME as variables, these are globals and already accessible in the scope of this enclosing function. Also we have a function sendDataToBothPlayers we can use, and we probably want to send the data to players no matter what.
Suggestions:
```
    const onShipExplode = (gameId) => {
        if (CLIENTS_HANDLER.areBothPlayersLoggedIn(gameId, AI_USERNAME, HUMAN_USERNAME)) {
            CLIENTS_HANDLER.updatePlayerStats(HUMAN_USERNAME, 'lose');
            CLIENTS_HANDLER.updatePlayerStats(AI_USERNAME, 'lose');
        }
        sendDataToBothPlayers(GameEndData('lose'));
        removeGame(gameId);
    }
```